edge-modules/DirectMethodCloudSender/src/Program.cs

```diff
@@ -2,119 +2,90 @@
 
-        public static int Main() => MainAsync().Result;
-
-        static async Task<int> MainAsync()
+        public static async Task Main()
         {
-            Logger.LogInformation("DirectMethodCloudSender Main() started.");
-
-            IConfiguration configuration = new ConfigurationBuilder()
-                .SetBasePath(Directory.GetCurrentDirectory())
-                .AddJsonFile("config/appsettings.json", optional: true)
-                .AddEnvironmentVariables()
-                .Build();
+            Logger.LogInformation($"Starting DirectMethodCloudSender with the following settings:\r\n{Settings.Current}");
 
-            string serviceClientConnectionString = Preconditions.CheckNonWhiteSpace(configuration.GetValue<string>("ServiceClientConnectionString"), nameof(serviceClientConnectionString));
-            // Get device id of this device, exposed as a system variable by the iot edge runtime
-            string targetDeviceId = configuration.GetValue<string>("IOTEDGE_DEVICEID");
-            string targetModuleId = configuration.GetValue("TargetModuleId", "DirectMethodReceiver");
-            TimeSpan dmDelay = configuration.GetValue("DirectMethodDelay", TimeSpan.FromSeconds(5));
-            Uri analyzerUrl = configuration.GetValue("AnalyzerUrl", new Uri("http://analyzer:15000"));
+            try
+            {
+                string serviceClientConnectionString = Settings.Current.ServiceClientConnectionString;
+                Uri analyzerUrl = Settings.Current.AnalyzerUrl;
 
-            AnalyzerClient analyzerClient = new AnalyzerClient { BaseUrl = analyzerUrl.AbsoluteUri };
+                ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(serviceClientConnectionString, (TransportType2)Settings.Current.TransportType);
+                AnalyzerClient analyzerClient = new AnalyzerClient { BaseUrl = analyzerUrl.AbsoluteUri };
 
-            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
+                (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
 
-            await CallDirectMethodFromCloud(serviceClientConnectionString, targetDeviceId, targetModuleId, dmDelay, analyzerClient, cts);
+                await CallDirectMethodFromCloud(serviceClient, Settings.Current.DirectMethodDelay, analyzerClient, cts);
 
-            completed.Set();
-            handler.ForEach(h => GC.KeepAlive(h));
-            Logger.LogInformation("DirectMethodCloudSender Main() finished.");
-            return 0;
+                completed.Set();
+                handler.ForEach(h => GC.KeepAlive(h));
+                Logger.LogInformation("DirectMethodCloudSender Main() finished.");
+            }
+            catch (Exception ex)
+            {
```

Comment:
I think we should try the exception because otherwise the initialization fails but the module keeps running and not doing anything. If it throws it will exit and then edgeAgent will retry to start

Reply:
I think I'm missing something. If initialization fails then won't the module enter this catch block then exit?